### PR TITLE
[@types/dc] updated dc type definitions to latest version

### PR DESCRIPTION
**Pull request by privateOmega**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by privateOmega** [comment]
@gordonwoodhull PTAL when you have bandwidth and I would really appreciate it if you could provide change suggestions on where you deem necessary.

File: types/dc/index.d.ts [hunk 1]
```diff
@@ -63,34 +64,42 @@ declare namespace dc {
...
 
     export interface Filter {
         isFiltered(value: any): boolean;
     }
 
     export interface Filters {
-        RangedFilter(low: any, high: any): Filter;
-        TwoDimensionalFilter(arr: Array<any>): Filter;
-        RangedTwoDimensionalFilter(arr: Array<any>): Filter;
+        RangedFilter(low: number, high: number): Array<number>;
```
Comment (hansrwindhoff) [inline types/dc/index.d.ts]: this seems incorrect, the return of these functions is an Array that has a isFiltered method, see the implementation:

the jsdoc annnotation in the code is incomplete...

/**
 * RangedFilter is a filter which accepts keys between `low` and `high`.  It is used to implement X
 * axis brushing for the {@link dc.coordinateGridMixin coordinate grid charts}.
 *
 * Its `filterType` is 'RangedFilter'
 * @name RangedFilter
 * @memberof dc.filters
 * @param {Number} low
 * @param {Number} high
 * @returns {Array<Number>}
 * @constructor
 */
dc.filters.RangedFilter = function (low, high) {
    var range = new Array(low, high);
    range.isFiltered = function (value) {
        return value >= this[0] && value < this[1];
    };
    range.filterType = 'RangedFilter';

    return range;
};

File: types/dc/index.d.ts [hunk 2]
```diff
@@ -63,34 +64,42 @@ declare namespace dc {
     }
 
     export interface Events {
-        trigger(fn: () => void, delay?: number): void;
+        trigger(closure: () => void, delay?: number): void;
```
Comment (hansrwindhoff) [inline types/dc/index.d.ts]: why the rename?

**Review by hansrwindhoff — CHANGES_REQUESTED** [review]

File: types/dc/index.d.ts (hunk 1 above)
Comment (privateOmega) [reply types/dc/index.d.ts]: @hansrwindhoff True, I just went off modifying the type definitions just based on jsdoc annotation and completely ignored the actual code. PTAL after my update.

**Comment by gordonwoodhull** [comment]
Hi @privateOmega. Thanks for putting in the effort here. The typings for dc.js have been severely out of date for a long time and I’ve tried to help a lot of people struggling with them.

Unfortunately I don’t currently use typescript and I think it would take a while for me to figure this out. There are also a number of edge cases like filters which I think may be difficult.

I definitely want to help, because this is important, but I’m not sure what to do.

File: types/dc/index.d.ts (hunk 2 above)
Comment (privateOmega) [reply types/dc/index.d.ts]: In [Line 1231](https://github.com/dc-js/dc.js/blob/56df1dcd3f67201b750ef989aaa1c3e15632096d/dc.js#L1231),  apparently the function argument was renamed.

**Comment by privateOmega** [comment]
@gordonwoodhull TBH this would be my first use of typescript per se, so I am not really acquainted with the best practices and all.
Also I updated the type definitions just based on jsdoc comments, there's a high chance some of it are wrong. :D

**Comment by privateOmega** [comment]
@hansrwindhoff It would be great if you could review the changes and merge the PR if there's no more issues with it.

**Review by hansrwindhoff — APPROVED** [review]
Thanks for the work!

**Comment by orta** [comment]
Sorry for the delay! Looks good to me 👍

**Pull request merged**